src/open_samus_returns_rando/multiworld_integration.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+
+from open_samus_returns_rando.files import files_path
+from open_samus_returns_rando.lua_editor import get_parent_for
+from open_samus_returns_rando.misc_patches.lua_util import lua_convert
+
+
+def get_lua_for_item(progression: list[list[dict[str, str | int]]]):
+    generic_pickup = """
+    Game.ImportLibrary("actors/items/randomizerpowerup/scripts/randomizerpowerup.lc", false)
+    MultiworldPickup = MultiworldPickup or {}
+    function MultiworldPickup.main()
+    end
+
+    function MultiworldPickup.OnPickedUp(progression, actorOrName)
+        RandomizerPowerup.OnPickedUp(progression, actorOrName)
+        hud.UpdatePlayerInfo(true)
+    end
+    """
+
+    main_item_id = str(progression[0][0]["item_id"])
+    parent = get_parent_for(main_item_id)
+    if parent == "RandomizerPowerup":
+        parent_content = generic_pickup
+    else:
+        parent_lower = parent.lower()
+        parent_content = files_path().joinpath("pickups", f"{parent_lower}.lua").read_text()
+        parent_content = parent_content.replace(parent, "MultiworldPickup")
+
+    progression_as_lua = lua_convert(progression, True)
+    return f'{parent_content}\nMultiworldPickup.OnPickedUp({progression_as_lua}, nil)'.replace("\n", "\\\n")
+
+
+def create_exefs_patches(output_exefs: Path, out_exheader: Path, input_exheader: Path, enabled: bool):
+    if not enabled:
+        return
+
+    import shutil
+
+    import ips
+
+    # Citra and Luma don't support patching the exheader. User needs to provide it as input and
+    # here the patch is just applied
+    exheader_ips_path = files_path().joinpath("exefs_patches", "exheader.ips")
+    with (
+          Path.open(exheader_ips_path, "rb") as exheader_ips,
+          Path.open(input_exheader, "rb") as original,
+          Path.open(out_exheader, "wb") as result
+        ):
+        content = exheader_ips.read()
+        patch = ips.Patch.load(content)
+        patch.apply(original, result)
+
+    # copy bps patch (don't ask me why the patch does not work as IPS format)
+    output_exefs.mkdir(parents=True, exist_ok=True)
+    shutil.copyfile(files_path().joinpath("exefs_patches", "code.bps"), output_exefs.joinpath("code.bps"))
```

Comment:
No idea if the patches works on NTSC and/or on Luma.
Code patch shouldn't be a problem for NTSC but no idea about exheader.
Have neither of them => Cannot test. If it doesn't work on Luma, I have no idea anyways on what to do.

On a side note: I was on Citra Canary 2704 and it had a bug in the services. Updated to Canary 2798 and it worked, so this will be the only version I will support.

Reply:
Conclusion is/was: 
Exheader needs to be provided by the user. Same patch works on NTSC and PAL. As I expect many users to input the wrong file, there is a stupid check function for it.
code patch can also use the same patch but Citra doesn't like the IPS format. We had to provide different BPS patch for NTSC and PAL but they do the same.

Atm no further work done on why it doesn't boot on luma.

Reply:
I should try this on luma now that we have the best patches